### PR TITLE
minor settings fixes

### PR DESCRIPTION
**Pull request by rbren**
**Short description of the problem this fixes or functionality that this introduces. This may be used for the CHANGELOG**

Minor follow up to https://github.com/All-Hands-AI/OpenHands/pull/3751. Makes sure that advanced settings show if you're using an unknown model

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

* show advanced settings if model is unknown
* add tests
* Fixes: https://github.com/All-Hands-AI/OpenHands/issues/3843

---
**Link of any specific issues this addresses**

### CODE REVIEW

**Comment by rbren** [comment]
CC @amanape and @tofarr --addressed your comments from https://github.com/All-Hands-AI/OpenHands/pull/3751

File: frontend/src/components/modals/settings/SettingsForm.tsx [hunk 1]
```diff
@@ -39,9 +39,11 @@ function SettingsForm({
...
   const { isOpen: isVisible, onOpenChange: onVisibleChange } = useDisclosure();
-  const advancedAlreadyInUse =
+  let advancedAlreadyInUse =
     !!settings.SECURITY_ANALYZER || !!settings.CONFIRMATION_MODE;
-  // TODO: || model is not in the list
+  if (settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)) {
+    advancedAlreadyInUse = true;
+  }
   const [enableAdvanced, setEnableAdvanced] =
     React.useState(advancedAlreadyInUse);
```
Comment (amanape) [inline frontend/src/components/modals/settings/SettingsForm.tsx:48]: Prefer using `React.useState` to ensure the component only re-renders when the state changes

```tsx
  // true if any advanced-related setting is already in use
  const [enableAdvanced, setEnableAdvanced] = React.useState(
    !!settings.SECURITY_ANALYZER ||
      !!settings.CONFIRMATION_MODE ||
      (!!settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)),
  );

  React.useEffect(() => {
    setEnableAdvanced(
      !!settings.SECURITY_ANALYZER ||
        !!settings.CONFIRMATION_MODE ||
        (!!settings.LLM_MODEL && !models.includes(settings.LLM_MODEL)),
    );
  }, [settings, models]);
```

Comment (rbren) [reply frontend/src/components/modals/settings/SettingsForm.tsx:48]: So I think we explicitly _don't_ want this useEffect--`advancedAlreadyInUse` should only be calculated once, when the component first renders, to decide what the initial view should be (advanced or not). After that the user is in control

Comment (rbren) [reply frontend/src/components/modals/settings/SettingsForm.tsx:48]: Also, for `useState`, I never fully understand React's magic here, but I _think_ you're allowed to set an intermediate variable (`advancedAlreadyInUse`) and pass that as the default to `useState`. Or am I mistaken?

Comment (rbren) [reply frontend/src/components/modals/settings/SettingsForm.tsx]: Thanks for the explanation! I think it's fixed now

**Review by amanape — APPROVED** [review]

**Pull request merged**